compiler/rustc_trait_selection/src/traits/structural_match.rs

```diff
@@ -247,12 +235,7 @@ impl<'a, 'tcx> TypeVisitor<'tcx> for Search<'a, 'tcx> {
         for field_ty in adt_def.all_fields().map(|field| field.ty(tcx, substs)) {
             let ty = self.tcx().normalize_erasing_regions(ty::ParamEnv::empty(), field_ty);
             debug!("structural-match ADT: field_ty={:?}, ty={:?}", field_ty, ty);
-
-            if ty.visit_with(self).is_break() {
-                // found an ADT without structural-match; halt visiting!
-                assert!(self.found.is_some());
-                return ControlFlow::BREAK;
-            }
+            ty.visit_with(self)?;
```

Comment:
nice

Reply:
I guess we could even use `try_for_each` here